### ISSUE TITLE
`invalid-formatter-suppression-comment (RUF028)` false positives

### ISSUE THREAD

**Issue by Avasam** [astral-sh/ruff#16104]
### Description

In https://github.com/Toufool/AutoSplit, I have a handful of false-positives for [invalid-formatter-suppression-comment (RUF028)](https://docs.astral.sh/ruff/rules/invalid-formatter-suppression-comment/#invalid-formatter-suppression-comment-ruf028):

```python
src\AutoSplit.py:550:53: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
548 |             or "Delayed Split" in self.current_split_image.text()
549 |             or not (
550 |                 self.skip_split_button.isEnabled()  # fmt: skip
    |                                                     ^^^^^^^^^^^ RUF028
551 |                 or self.is_auto_controlled
552 |                 or navigate_image_only
    |
    = help: Remove this comment

src\AutoSplit.py:962:29: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
960 |         # Get split image
961 |         self.split_image = (
962 |             specific_image  # fmt: skip
    |                             ^^^^^^^^^^^ RUF028
963 |             or self.split_images_and_loop_number[0 + self.split_image_number][0]
964 |         )
    |
    = help: Remove this comment

src\capture_method\DesktopDuplicationCaptureMethod.py:51:46: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
   |
49 |             display
50 |             for display in self.desktop_duplication.displays
51 |             if display.hmonitor == hmonitor  # fmt: skip
   |                                              ^^^^^^^^^^^ RUF028
52 |         )
53 |         offset_x, offset_y, *_ = win32gui.GetWindowRect(hwnd)
   |
   = help: Remove this comment

src\capture_method\WindowsGraphicsCaptureMethod.py:165:54: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
163 |     def check_selected_region_exists(self):
164 |         return bool(
165 |             is_valid_hwnd(self._autosplit_ref.hwnd)  # fmt: skip
    |                                                      ^^^^^^^^^^^ RUF028
166 |             and self.frame_pool
167 |             and self.session
    |
    = help: Remove this comment

src\compare.py:85:42: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
   |
83 |     # that the value can be. Used for normalizing from 0 to 1.
84 |     max_error = (
85 |         source.size * MAXBYTE * MAXBYTE  # fmt: skip
   |                                          ^^^^^^^^^^^ RUF028
86 |         if not is_valid_image(mask)
87 |         else cv2.countNonZero(mask)
   |
   = help: Remove this comment

src\hotkeys.py:139:40: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
137 |     # keyboard.send(keyboard.key_to_scan_codes(key_or_scan_code)[1])
138 |     pyautogui.hotkey(*[
139 |         "+" if key == "plus" else key  # fmt: skip
    |                                        ^^^^^^^^^^^ RUF028
140 |         for key in hotkey_or_scan_code.replace(" ", "").split("+")
141 |     ])
    |
    = help: Remove this comment

src\menu_bar.py:190:9: RUF028 This suppression comment is invalid because it cannot be on its own line
    |
188 |         self.setFocus()
189 |
190 |         # region Build the Capture method combobox  # fmt: skip
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ RUF028
191 |
192 |         capture_method_values = CAPTURE_METHODS.values()
    |
    = help: Remove this comment

src\menu_bar.py:239:35: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
237 |         try:
238 |             return [
239 |                 device.device_id  # fmt: skip
    |                                   ^^^^^^^^^^^ RUF028
240 |                 for device in self.__video_capture_devices
241 |             ].index(capture_device_id)
    |
    = help: Remove this comment

src\user_profile.py:194:34: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
192 |     )
193 |     if not (
194 |         load_settings_file_path  # fmt: skip
    |                                  ^^^^^^^^^^^ RUF028
195 |         and __load_settings_from_file(autosplit, load_settings_file_path)
196 |     ):
    |
    = help: Remove this comment

src\user_profile.py:207:15: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
    |
205 | def load_settings_on_open(autosplit: "AutoSplit"):
206 |     settings_files = [
207 |         file  # fmt: skip
    |               ^^^^^^^^^^^ RUF028
208 |         for file in os.listdir(auto_split_directory)
209 |         if file.endswith(".toml")
    |
    = help: Remove this comment
```

Removing any of them would cause the formatter (`ruff format`) to unwrap the line. These are all related to https://github.com/astral-sh/ruff/issues/5528#issuecomment-2564419176 and https://github.com/astral-sh/ruff/issues/11753#issuecomment-2161337125

Ruff version: 0.9.6

Settings: [ruff.toml.txt](https://github.com/user-attachments/files/18758584/ruff.toml.txt)

**Comment by MichaReiser** [comment]
These are all true positives with the exception of:

```
src\capture_method\DesktopDuplicationCaptureMethod.py:51:46: RUF028 This suppression comment is invalid because it cannot be in an expression, pattern, argument list, or other non-statement
   |
49 |             display
50 |             for display in self.desktop_duplication.displays
51 |             if display.hmonitor == hmonitor  # fmt: skip
   |                                              ^^^^^^^^^^^ RUF028
52 |         )
53 |         offset_x, offset_y, *_ = win32gui.GetWindowRect(hwnd)
   |
   = help: Remove this comment
```


The reason why the formatter doesn't collapse the other expressions is because there's an end-of-line comment inside the expression but it could also be `# blabla` instead of `fmt: skip`

**Comment by MichaReiser** [comment]
Actually, never mind. All of those are true-positives. I thought the one is an if-statement but it's also an if expression.

**Comment by Avasam** [comment]
Oh I see, having any comment there would work.
But I do use it to prevent formatter from formatting, it's be weird to use any other comment than `# fmt: skip`.

I'm not sure what to make of this, I'll just keep `RUF028` disabled for now.

**Comment by MichaReiser** [comment]
>  But I do use it to prevent formatter from formatting, it's be weird to use any other comment than # fmt: skip.

That makes sense. You could do `fmt: ignore`,  `fmt: split`, `fmt: splitting on multiple lines is prettier` or any other comment for as long as it isn't `fmt: off` or `fmt: skip` because the formatter will keep formatting it. 

> Would it make sense to request my usage to be supported? (ie actually telling the formatter to not unwrap, rather than just incidentally not unwrapping because of an arbitrary comment)

I don't think that's unreasonable. I considered it when working on suppression comments. There are two reasons why I decided against it:

1. I found it unintuitive how and when and where expression level suppression comments apply in Black. [discussion](https://github.com/astral-sh/ruff/discussions/6338)
2. Implementing suppressions is fairly complicated in our design and I rather spent the time on improving the formatting itself, than making suppressions better. However, we could consider supporting a suppression comment similar to Prettier where you add the suppression before the expression you want to suppress rather than at its end (https://prettier.io/docs/ignore.html#javascript). However, Python is a bit different than JS because it sometimes requires parentheses around expressions to prevent syntax errors where JS doesn't.

I hope that makes sense to you.
